internal/oci/container.go

```diff
@@ -448,13 +460,30 @@ func (c *Container) verifyPid() error {
 }
 
 // getPidStartTime reads the kernel's /proc entry for stime for PID.
-func getPidStartTime(pid int) (int, error) {
-	var st unix.Stat_t
-	if err := unix.Stat(fmt.Sprintf("/proc/%d", pid), &st); err != nil {
-		return 0, errors.Wrapf(ErrNotFound, err.Error())
+// inspiration for this function came from https://github.com/containers/psgo/blob/master/internal/proc/stat.go
+// some credit goes to the psgo authors
+func getPidStartTime(pid int) (string, error) {
+	data, err := ioutil.ReadFile(fmt.Sprintf("/proc/%d/stat", pid))
+	if err != nil {
+		return "", errors.Wrapf(ErrNotFound, err.Error())
+	}
+	// The command (2nd field) is wrapped in ().
+	// Filter it out
+	commEnd := strings.LastIndexByte(string(data), ')')
+	if commEnd == -1 {
+		return "", errors.Wrapf(ErrNotFound, "invalid format in stat file")
+	}
+
+	// commEnd now points to the ending parenthesis.
+	// After it, there is a space, and then the next field starts
+	statFieldsAfterComm := strings.Split(string(data[commEnd+2:]), " ")
```

Comment:
I suggest you don't split the whole string, but count 20 spaces from the position you found and then just take the number. Otherwise you're creating a slice of 50+ fields of which 49 you won't need.

Reply:
done!